hoomd/md/manifold.py

```diff
@@ -0,0 +1,389 @@
+    manifolds are described by implicit functions.
+
+    Attention:
+        Users should not instantiate :py:class:`Manifold` directly.
+
+    Warning:
+        Only one manifold can be applied to the methods/active forces.
+
+    """
+    def __init__(self):
+
+        self._cpp_obj = None
+
+        self.name = None
+
+
+class Cylinder(Manifold):
+    R""" Cylinder manifold.
+
+    Args:
+        r (`float`): radius of the cylinder constraint (in distance units).
+        P (`tuple`[ `float`, `float`, `float`]): point defining position of the cylinder axis (default origin).
+
+    :py:class:`Cylinder` specifies that a cylindric manifold is defined as
+    a constraint.
+
+    Note:
+        The cylinder axis is parallel to the z-direction.
+
+    .. rubric:: Implicit function
+
+    .. math::
+        F(x,y,z) = x^{2} + y^{2} - r^{2}
+
+    Example::
+
+        cylinder1 = manifold.Cylinder(r=10)
+        cylinder2 = manifold.Cylinder(r=5,P=(1,1,1))
+    """
+    def __init__(self,r, P=[0,0,0] ):
+        # initialize the base class
+        super().__init__();
+        param_dict = ParameterDict(
+            r=float(r),
+            P=(float, float,float),
+        )
+        param_dict.update(
+            dict(P=(P[0], P[1], P[2])))
+
+        self._param_dict.update(param_dict)
+        # set defaults
+
+    def _attach(self):
+        self._cpp_obj = _md.ManifoldCylinder(self.r, _hoomd.make_scalar3( self.P[0], self.P[1], self.P[2]) );
+
+        self.name = "Cylinder"
+
+        super()._attach()
+
+class Diamond(Manifold):
+
+    R""" Triply periodic diamond manifold.
+
+    Args:
+        N (`tuple`[`int`, `int`, `int`] or `int`): number of unit cells in all 3 directions.
+            :math:`[N_x, N_y, N_z]`. In case number of unit cells u in all
+            direction the same (:math:`[u, u, u]`), use ``N = u``.
+        epsilon (`float`): defines CMC companion of the Diamond surface (default 0) 
+
+    :py:class:`Diamond` specifies a periodic diamond surface as a constraint.
+    The diamond (or Schwarz D) belongs to the family of triply periodic minimal surfaces.
+
+    
+    For the diamond surface, see:
+    * `A. H. Schoen 1970  <https://ntrs.nasa.gov/citations/19700020472>`_
+    * `P. J. F. Gandy et. al. 1999  <https://doi.org/10.1016/S0009-2614(99)01000-3>`_
+    * `H. G. von Schnering and R. Nesper 1991  <https://doi.org/10.1007/BF01313411>`_
+    
+    .. rubric:: Implicit function
+    
+    .. math::
+        F(x,y,z) = \cos{\frac{2 \pi}{L_x} x}\cdot\cos{\frac{2 \pi}{L_y} y}\cdot\cos{\frac{2 \pi}{L_z} z} - \sin{\frac{2 \pi}{L_x} x}\cdot\sin{\frac{2 \pi}{L_y} y}\cdot\sin{\frac{2 \pi}{L_z} z}
+
+    is the nodal approximation of the diamond surface where :math:`[L_x,L_y,L_z]` is the periodicity length in the x, y and z direction.
+    The periodicity length L is defined by the box size B and the number of unit cells N. :math:`L=\frac{B}{N}`
```

Comment:
```suggestion
    The periodicity length L is defined by the current box size B and the number of unit cells N. :math:`L=\frac{B}{N}`
```

Reply:
Or maybe add a more explicit note that the manifold is always scaled to fit inside the current box. The same comment applies for Gyroid and the other manifolds that are scaled.